hal/README.md

```diff
@@ -134,11 +134,41 @@ The object provided to the Mustache template for automatically-generated documen
 }
 ```
 
+## Utilities `@azure-iot/hal/util`
+
+### Server
+Server-specific utility methods can be accessed by calling the utility import as a function on the server object, eg. `util(server).method(...)`.
```

Comment:
nit: I'm assuming that "method" in `util(server).method(...)` is one of the methods below. maybe `util(server).<method>(...)` would be less confusing since it's (hopefully) clear that you can't execute that as is